### DISCUSSION TITLE
Sign In With Apple when running on iOS

### DISCUSSION THREAD

**Discussion by thecoolwinter** [supabase/supabase#2805] [General]
When using Sign in with Apple (SIWA) from native swift running on iOS, users expect to be shown the native log in UI instead of doing the OAuth redirect. This native sign in returns an object with the parameters:
- identityToken (JWT access token)
- authorizationCode (refresh token)
- name and email and scopes

When GoTrue signs users in using SIWA they're redirected to the OAuth apple site where they sign in and eventually GoTrue received the above parameters, just through the OAuth callback flow. If this flow were to be used on iOS, it would be a very clunky UX, and could be grounds for an iOS app to be rejected from review. 

My questions are thus:
- **Is it possible to send the above parameters to GoTrue and create/sign in a user?** *Eg: use the native SIWA flow and send the received object instead of going through the OAuth flow*
- **Is that solution safe to use in production?**

**Comment by catlan** [comment]
I started playing with this a bit: https://github.com/supabase/gotrue-swift/issues/4

**Reply by johndpope** [reply]
I have updated this ticket with sample nodejs repo to issue jwt on user sign up from authorisation code. Needs more work.

**Comment by Xanewok** [comment]
I believe https://github.com/supabase/gotrue/pull/689 (built on https://github.com/supabase/gotrue/pull/189) addresses this and allows to sign-in the user with the ID token for hosted instances.
